beacon-chain/rpc/service.go

```diff
@@ -207,50 +207,88 @@ func (s *Service) CurrentAssignmentsAndGenesisTime(
 
 // ProposeBlock is called by a proposer in a sharding validator and a full beacon node
 // sends the request into a beacon block that can then be included in a canonical chain.
-func (s *Service) ProposeBlock(ctx context.Context, req *pb.ProposeRequest) (*pb.ProposeResponse, error) {
+func (s *Service) ProposeBlock(ctx context.Context, blk *pbp2p.BeaconBlock) (*pb.ProposeResponse, error) {
+
+	h, err := b.Hash(blk)
+	if err != nil {
+		return nil, fmt.Errorf("could not hash block: %v", err)
+	}
+	log.WithField("blockHash", fmt.Sprintf("%#x", h)).Debugf("Block proposal received via RPC")
+	// We relay the received block from the proposer to the chain service for processing.
+	s.chainService.IncomingBlockFeed().Send(blk)
+	return &pb.ProposeResponse{BlockHash: h[:]}, nil
+}
+
+// CurrentPOWChainBlockHash retrieves the latest blockhash of the POW chain and sends it to the validator client.
+func (s *Service) CurrentPOWChainBlockHash(ctx context.Context, req *ptypes.Empty) (*pb.POWChainResponse, error) {
 	var powChainHash common.Hash
+
 	if !s.enablePOWChain {
-		powChainHash = common.BytesToHash([]byte{byte(req.SlotNumber)})
-	} else {
-		powChainHash = s.powChainService.LatestBlockHash()
+		powChainHash = common.BytesToHash([]byte{'p', 'o', 'w', 'c', 'h', 'a', 'i', 'n'})
+
+		return &pb.POWChainResponse{
+			BlockHash: powChainHash[:],
+		}, nil
 	}
 
-	//TODO(#589) The attestation should be aggregated in the validator client side not in the beacon node.
+	powChainHash = s.powChainService.LatestBlockHash()
+
+	return &pb.POWChainResponse{
+		BlockHash: powChainHash[:],
+	}, nil
+
+}
+
+// ComputeStateRootForBlock computes the state root after a block has been processed through a state transition and
```

Comment:
I think we can shorten the name to `ComputeStateRoot` since we don't compute state root for anything else except block